pkg/image/prune/imagepruner.go

```diff
@@ -53,12 +67,59 @@ type ImagePruner interface {
 
 // imagePruner implements ImagePruner.
 type imagePruner struct {
-	g         graph.Graph
-	algorithm pruneAlgorithm
+	g              graph.Graph
+	algorithm      pruneAlgorithm
+	registryPinger registryPinger
 }
 
 var _ ImagePruner = &imagePruner{}
 
+// registryPinger performs a health check against a registry.
+type registryPinger interface {
+	// ping performs a health check against registry.
+	ping(registry string) error
+}
+
+// defaultRegistryPinger implements registryPinger.
+type defaultRegistryPinger struct {
+}
+
+func (drp *defaultRegistryPinger) ping(registry string) error {
+	healthzCheck := func(proto, registry string) error {
+		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))
```

Comment:
That's exactly what this is doing - failing the prune if the registry can't be contacted.